cadence/contracts/bridge/FlowEVMBridgeCustomAssociationTypes.cdc

```diff
@@ -0,0 +1,149 @@
+import "NonFungibleToken"
+import "CrossVMMetadataViews"
+import "EVM"
+
+/// This contract defines types required for custom cross-VM associations as used in FlowEVMBridgeCustomAssociation
+/// and in EVM-native cross-VM NFTs.
+///
+access(all) contract FlowEVMBridgeCustomAssociationTypes {
+
+    access(all) entitlement FulfillFromEVM
+
+    /// Resource interface used by EVM-native NFT collections allowing for the fulfillment of NFTs from EVM into Cadence
+    ///
+    access(all) resource interface NFTFulfillmentMinter {
+        /// Getter for the type of NFT that's fulfilled by this implementation
+        ///
+        access(all) view fun getFulfilledType(): Type
+
+        /// Called by the VM bridge when moving NFTs from EVM into Cadence if the NFT is not in escrow. Since such NFTs
+        /// are EVM-native, they are distributed in EVM. On the Cadence side, those NFTs are handled by a mint & escrow
+        /// pattern. On moving to EVM, the NFTs are minted if not in escrow at the time of bridging.
+        ///
+        /// @param id: The id of the token being fulfilled from EVM
+        ///
+        /// @return The NFT fulfilled from EVM as its Cadence implementation
+        ///
+        access(FulfillFromEVM)
+        fun fulfillFromEVM(id: UInt256): @{NonFungibleToken.NFT} {
+            pre {
+                id <= UInt256(UInt64.max):
+                "The requested ID \(id.toString()) exceeds the maximum assignable Cadence NFT ID \(UInt64.max.toString())"
+            }
+            post {
+                UInt256(result.id) == id:
+                "Resulting NFT ID \(result.id.toString()) does not match requested ID \(id.toString())"
+                result.getType() == self.getFulfilledType():
+                "Expected \(self.getFulfilledType().identifier) but fulfilled \(result.getType().identifier)"
+            }
+        }
+    }
+
+    /// Resource interface retrieving all relevant information for the VM bridge to fulfill bridge requests. The
+    /// interface allows for extensibility in the event future config types are added in the future for various asset
+    /// types.
+    ///
+    access(all) resource interface CustomConfig {
+        access(all) view fun getCadenceType(): Type
```

Comment:
Maybe we could have a function that gets the interface types to show what standard it conforms to? like this?
```
access(all) view fun getCadenceStandardInterfaceTypes(): [Type]
```
and it can return `{NonFungibleToken.NFT}` and/or `{FungibleToken.Vault}` and/or whatever other token standards people come up with in the future

Reply:
Added in [db8ecce](https://github.com/onflow/flow-evm-bridge/pull/173/commits/db8eccead5836f118a7b2269fa98cf2f4815aa06). I mirrored the `NonFungibleToken.Receiver.getSupportedTypes` return type of `{Type: Bool}`. Lmk if you still think the `[Type]` return value makes more sense here

Reply:
I think the dictionary type is a good idea. 👍